### PR TITLE
Tiling Problem 2XN

### PR DESCRIPTION
**Pull request by yashmita**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have added the code for tiling problem using C language. The problem is to find the number of ways in which the floor of size 2XN can be tiled using tiles of size 2X1. 

### Checklist

- [x] I've read the contribution guidelines. 
- [x] I've checked the issue list before deciding what to submit. 
- [x] I've edited the `README.md` and link to my code. 

## Related Issues or Pull Requests

This is the PR for issue #4430

### CODE REVIEW

File: C/dp/tiling_problem.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+PROBLEM STATEMENT:
+Given a floor of size 2Xn and tiles of size 2X1. We've to find the number number of ways in which the floor can be tiled using tiles of size 2X1. 
+A tile can be either placed horizontally or vertically.
+*/
+
+#include<stdio.h>
+
+//A function that uses DP approach to find the total number of ways to tile the floor.
+int NoOfWays(int n){
+	//An array, tiles of size n+1 which stores the no. of ways in which the floor of size 2Xi can be tiled
```
Comment (HarshCasper) [inline C/dp/tiling_problem.c]: Use comments when necessary

File: C/dp/tiling_problem.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	return tiles[n];
+}
+
+//Driver function
+int main(){
+	int n;
+	printf("Enter value of n: ");
+	//user input for value of n
+	scanf("%d",&n);
+	//function call to NoOfWays
```
Comment (HarshCasper) [inline C/dp/tiling_problem.c]: Use comments when necessary

File: C/dp/tiling_problem.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	}
+	//returning the ways in which floor of size 2Xn can be tiled
+	return tiles[n];
+}
+
+//Driver function
+int main(){
+	int n;
+	printf("Enter value of n: ");
+	//user input for value of n
```
Comment (HarshCasper) [inline C/dp/tiling_problem.c]: Use comments when necessary

File: C/dp/tiling_problem.c [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+Input: 4
+Output: 3
+
+2.
+Input: 1
+Output: 1
+
+3.
+Input: 8
+Output: 21
```
Comment (HarshCasper) [inline C/dp/tiling_problem.c]: Remove these

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by yashmita** [comment]
I have updated the changes as requested. Kindly have a look at it.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**